src/Engine/ProtoCore/RuntimeData.cs

```diff
@@ -120,6 +125,15 @@ public CallSite GetCallSite(int classScope, string methodName, Executable execut
 
                 CallsiteCache[callsiteID] = csInstance;
                 CallSiteToNodeMap[csInstance.CallSiteID] = topGraphNode.guid;
+                List<string> callsiteIDs;
+                if (NodeToCallsiteIdentifiersMap.TryGetValue(topGraphNode.guid, out callsiteIDs))
+                {
+                    callsiteIDs.Add(callsiteID);
+                }
+                else
+                {
+                    NodeToCallsiteIdentifiersMap[topGraphNode.guid] = new List<string>() { callsiteID };
```

Comment:
Why can't we simply declare a `NodeToCallsiteMap` as:
```
Dictionary<Guid, List<CallSite>> NodeToCallsiteMap;
```
Then we can simply say:
```
NodeToCallsiteMap[topGraphNode.guid] = new List<CallSite>() { csInstance };
```

Reply:
Yes we can do that instead.... I wasn't sure the implications of having two maps that have Callsite Values.  Seems ok since the CallsiteCache is never modified other than `add`